### PR TITLE
adaptations to support Kibana 7.8.0

### PR DESCRIPTION
**Pull request by darkrst**
Changes to migrate the plugin to Kibana 7.8

### CODE REVIEW

File: public/data_load/enhanced-table-request-handler.js [hunk 1]
```diff
@@ -23,12 +23,16 @@ import { SearchSource } from '../../../../src/plugins/data/public/search/search_
 import { RequestAdapter, DataAdapter } from '../../../../src/plugins/inspector/public';
 import { getSearchService, getQueryService } from '../../../../src/plugins/data/public/services';
 import { serializeAggConfig } from '../../../../src/plugins/data/public/search/expressions/utils';
+import { npStart } from '../legacy_imports';
+import { search } from '../../../../src/plugins/data/public'
```
Comment (fbaligand) [inline public/data_load/enhanced-table-request-handler.js:27]: It seems that this dependency (search) is not used?

File: public/legacy.ts [hunk 2]
```diff
@@ -17,20 +17,27 @@
...
+import { PluginInitializerContext  } from '../../../src/core/public';
 import { npSetup, npStart } from './legacy_imports';
 import { plugin } from '.';
 
 import { TablePluginSetupDependencies } from './plugin';
-import { setup as visualizationsSetup } from '../../../src/legacy/core_plugins/visualizations/public/np_ready/public/legacy';
+import { TablePluginStartDependencies } from './plugin';
+import { DataPublicPluginStart } from '../../../src/plugins/data/public';
+//import { plugin as VisualizationsSetup } from '../../../src/plugins/visualizations/public/plugin';
+//import { setup as visualizationsSetup } from '../../../src/plugins/visualizations/public/legacy';
```
Comment (fbaligand) [inline public/legacy.ts:28]: could you remove these commented lines?

File: public/vis_controller.ts [hunk 3]
```diff
@@ -20,86 +20,93 @@
 import angular, { IModule, auto, IRootScopeService, IScope, ICompileService } from 'angular';
 import $ from 'jquery';
 
-import { VisParams } from 'src/plugins/visualizations/public';
-import { npStart } from './legacy_imports';
+import { CoreSetup, PluginInitializerContext } from 'kibana/public';
+import { VisParams, ExprVis } from '../../../src/plugins/visualizations/public';
+//import { npStart } from './legacy_imports';
```
Comment (fbaligand) [inline public/vis_controller.ts:25]: Could you remove this commented line?

**Comment by fbaligand** [comment]
As core table_vis_type, could you remove root `index.ts` file and add `kibana.json`?
Bringing compatibility with last Kibana version is also the opportunity to migrate more to Kibana new platform.

**Comment by darkrst** [comment]
> As core table_vis_type, could you remove root `index.ts` file and add `kibana.json`?
> Bringing compatibility with last Kibana version is also the opportunity to migrate more to Kibana new platform.

There seems to be an issue with just replacing index.ts for kibana.json - in the process of starting, it's trying to register the plugin twice and throws this error:
 FATAL  Error: Plugin with id "enhancedTable" is already registered!

Have you experienced this before?

**Comment by fbaligand** [comment]
No, I have not experimented this kind of error.
I managed to avoid this error, doing "yarn start" in kibana home folder (not in enhanced-table folder).
But then, I got tons of errors saying globally that my plugin has still legacy dependencies.
So it seems that enhanced table plugin is not ready yet to use "kibana.json".
This will for another time...
So could you go back to previous state: no kibana.json and instead index.ts?

**Comment by darkrst** [comment]
Yes, just updated it.

File: public/document-table-vis.js [hunk 4]
```diff
@@ -18,104 +18,113 @@
...
 
 import tableVisTemplate from './enhanced-table-vis.html';
-import { EnhancedTableVisualizationController } from './vis_controller';
+import { getEnhancedTableVisualizationController } from './vis_controller';
 import { enhancedTableRequestHandler } from './data_load/enhanced-table-request-handler';
 import { documentTableResponseHandler } from './data_load/document-table-response-handler';
 import { DocumentTableData } from './components/document_table_vis_data';
 import { EnhancedTableOptions } from './components/enhanced_table_vis_options';
 
+const toExpression = (vis) => {
```
Comment (fbaligand) [inline public/document-table-vis.js:31]: Could you replace this line by:
`function toExpression(vis) {`

As it is a global function, I prefer this form.

File: public/document-table-vis.js [hunk 5]
```diff
@@ -18,104 +18,113 @@
...
+  let pipeline = `kibana | enhanced_table_visualization type='${vis.type.name}'
+    ${prepareJson('visConfig', visConfig)}
+    ${prepareJson('aggConfigs', aggs.aggs)}
+    metricsAtAllLevels=${vis.isHierarchical()}
+    partialRows=${vis.type.requiresPartialRows || vis.params.showPartialRows || false} `;
+  if (indexPattern) {
+    pipeline += `${prepareString('index', indexPattern.id)}`;
+  }
+  return pipeline;
+};
```
Comment (fbaligand) [inline public/document-table-vis.js:43]: could you remove trailing `;`?

this is because of my previous change request :)

File: public/enhanced-table-vis.js [hunk 6]
```diff
@@ -19,117 +19,126 @@
...
+import { prepareJson, prepareString } from '../../../src/plugins/visualizations/public/legacy/build_pipeline';
 
 import tableVisTemplate from './enhanced-table-vis.html';
-import { EnhancedTableVisualizationController } from './vis_controller';
+import { getEnhancedTableVisualizationController } from './vis_controller';
 import { enhancedTableRequestHandler } from './data_load/enhanced-table-request-handler';
 import { enhancedTableResponseHandler } from './data_load/enhanced-table-response-handler';
 import { EnhancedTableOptions } from './components/enhanced_table_vis_options';
 
+const toExpression = (vis) => {
```
Comment (fbaligand) [inline public/enhanced-table-vis.js:31]: Same request here, could you replace the line by:
`function toExpression(vis) {`

File: public/enhanced-table-vis.js [hunk 7]
```diff
@@ -19,117 +19,126 @@
...
+  let pipeline = `kibana | enhanced_table_visualization type='${vis.type.name}'
+    ${prepareJson('visConfig', visConfig)}
+    ${prepareJson('aggConfigs', aggs.aggs)}
+    metricsAtAllLevels=${vis.isHierarchical()}
+    partialRows=${vis.type.requiresPartialRows || vis.params.showPartialRows || false} `;
+  if (indexPattern) {
+    pipeline += `${prepareString('index', indexPattern.id)}`;
+  }
+  return pipeline;
+};
```
Comment (fbaligand) [inline public/enhanced-table-vis.js:43]: could you remove trailing `;` here?

**Comment by darkrst** [comment]
All changes applied! :)

File: public/document-table-vis.js [hunk 8]
```diff
@@ -18,104 +18,113 @@
...
+import { getEnhancedTableVisualizationController } from './vis_controller';
 import { enhancedTableRequestHandler } from './data_load/enhanced-table-request-handler';
 import { documentTableResponseHandler } from './data_load/document-table-response-handler';
 import { DocumentTableData } from './components/document_table_vis_data';
 import { EnhancedTableOptions } from './components/enhanced_table_vis_options';
 
+function toExpression(vis) {
+  const visConfig = { ...vis.params };
+  const { indexPattern, aggs } = vis.data;
+  let pipeline = `kibana | enhanced_table_visualization type='${vis.type.name}'
```
Comment (fbaligand) [inline public/document-table-vis.js:34]: Why did you add `kibana |` in this line?
I just tested without this prefix, and it works great.
I prefer to have the minimum pipeline expression, and `kibana` item is already put in expression by `build_pipeline.ts`

File: public/enhanced-table-vis.js [hunk 9]
```diff
@@ -19,117 +19,126 @@
...
-import { EnhancedTableVisualizationController } from './vis_controller';
+import { getEnhancedTableVisualizationController } from './vis_controller';
 import { enhancedTableRequestHandler } from './data_load/enhanced-table-request-handler';
 import { enhancedTableResponseHandler } from './data_load/enhanced-table-response-handler';
 import { EnhancedTableOptions } from './components/enhanced_table_vis_options';
 
+function toExpression(vis) {
+  const visConfig = { ...vis.params };
+  const { indexPattern, aggs } = vis.data;
+  let pipeline = `kibana | enhanced_table_visualization type='${vis.type.name}'
```
Comment (fbaligand) [inline public/enhanced-table-vis.js:34]: same remark here.

**Comment by fbaligand** [comment]
I had done more tests on my computer, before to merge.
And that's all right.
Sorry for the one day delay to give you feedback.
I just added a last request change on document-table-vis and enhanced-table-vis.
After that, I promise that I merge the PR.

File: public/document-table-vis.js (hunk 8 above)
Comment (darkrst) [reply public/document-table-vis.js:34]: I think at one point I was under the impression that the pipeline variable gets overwritten instead of appended to, as that's how it came out in the compiled code, but now that I look at the actual .ts files I realize that's not the case. I'm removing those  and pushing the changes in a minute.

File: public/enhanced-table-vis.js (hunk 9 above)
Comment (darkrst) [reply public/enhanced-table-vis.js]: Fixed

**Comment by darkrst** [comment]
All set, those two are fixed now. Thanks for taking the time to do tests.

**Comment by fbaligand** [comment]
Nice!
Thank you for this last fix.

**Comment by fbaligand** [comment]
So I have only a last word to say: LGTM!
